src/foam/parse/FObjectParser.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.parse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import foam.core.X;
+import foam.lib.parse.Alt;
+import foam.lib.parse.PStream;
+import foam.lib.parse.Parser;
+import foam.lib.parse.ParserContext;
+import foam.lib.parse.ProxyParser;
+import foam.mlang.predicate.Predicate;
+
+public class FObjectParser extends ProxyParser {

```

Comment:
Again, can we use the one in foam/lib/json/FObjectParser.java ?  Are there any differences here that we need?